packages/routing-utils/src/types.ts

```diff
@@ -67,17 +72,32 @@ export interface VercelConfig {
 export interface Rewrite {
   source: string;
   destination: string;
+  has?: Array<{
+    type: 'header' | 'cookie' | 'query';
+    key: string;
+    value?: string;
+  }>;
 }
 
 export interface Redirect {
   source: string;
   destination: string;
   permanent?: boolean;
   statusCode?: number;
+  has?: Array<{
+    type: 'header' | 'cookie' | 'query';
+    key: string;
+    value?: string;
+  }>;
 }
 
 export interface Header {
   source: string;
+  has?: Array<{
+    type: 'header' | 'cookie' | 'query';
+    key: string;
+    value?: string;
+  }>;
```

Comment:
If these are all the same, then it might make more sense to create a new type

```ts
export interface HasKeyValue {
    type: 'header' | 'cookie' | 'query';
    key: string;
    value?: string;
  }
```

Reply:
Or maybe just

```ts
export type Has = Array<{
    type: 'header' | 'cookie' | 'query';
    key: string;
    value?: string;
  }>
```

Reply:
Updated to share the type 👍